src/shared/main/java/com/couchbase/lite/internal/support/Log.java

```diff
@@ -408,6 +332,32 @@ public static void setC4LogLevel(EnumSet<LogDomain> domains, LogLevel level) {
         }
     }
 
+    private static void log(
+        @NonNull LogLevel level,
+        @NonNull LogDomain domain,
+        @Nullable Throwable err,
+        @Nullable String msg,
+        Object... args) {
+        String message = lookupErrorMessage(msg);
+
+        if ((args != null) && (args.length > 0)) { message = formatMessage(message, args); }
+
+        if (err != null) { message = message + " (" + err + ")"; }
+
+        sendToLoggers(level, domain, message);
+    }
+
+    private static String lookupErrorMessage(String error) {
+        final String message = errorMessages.get(error);
+        return (message == null) ? error : message;
+    }
+
+    private static String formatMessage(String msg, Object... args) {
+        try { return String.format(msg, (Object[]) args); }
+        catch (IllegalFormatException | FormatterClosedException ignore) { }
```

Comment:
not seeing the Locale.ENGLISH setup, also exceptions are not handled. is this intentional?

Reply:
Good call on the Local.  Will fix.  Yes.  If there is a formatting exception, the code will simply return the value of `msg` without trying to fill in the template.  I'm thinking that's the best I can do...